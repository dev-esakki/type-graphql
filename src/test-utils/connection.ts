import { createConnection } from 'typeorm';

export const testConnection = (drop: boolean = false) => {
    return createConnection(
        {
            "type": "mysql",
            "host": "localhost",
            "port": 3306,
            "username": "root",
            "password": "Trio@123",
            "database": "test",
            "synchronize": true,
            "logging": false,
            "dropSchema": drop,
            "entities": [
                __dirname + "/../entity/*.*"
               //"src/entity/**/*.ts"
            ],
            "migrations": [
               "src/migration/**/*.ts"
            ],
            "subscribers": [
               "src/subscriber/**/*.ts"
            ],
            "cli": {
               "entitiesDir": "src/entity",
               "migrationsDir": "src/migration",
               "subscribersDir": "src/subscriber"
            }
         }
    )
}