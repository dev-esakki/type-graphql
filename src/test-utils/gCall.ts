import { gqSchema } from './../schema';


import { graphql, GraphQLSchema } from 'graphql'
import { Maybe } from 'type-graphql';

interface Options {
    source: string //graphql schema query or mutation
    variableValues?: Maybe<{ // graphql variables it may be optional also
        [key: string]: any
    }>
    userid?: any
}

let schema: GraphQLSchema;

export const gCall = async({ source, variableValues, userid }: Options) => {
    if(!schema) {
        schema = await gqSchema();
    }
    return graphql({
        schema,
        source,
        variableValues,
        contextValue: {
            req: {
                headers: {
                    userid
                }
            },
        }
    })
}