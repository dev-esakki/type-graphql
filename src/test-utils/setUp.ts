import { testConnection } from './connection';

testConnection(false).then(() => process.exit());