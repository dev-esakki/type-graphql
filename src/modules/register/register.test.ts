import { gCall } from './../../test-utils/gCall';
import faker from 'faker'
import { testConnection } from './../../test-utils/connection';
import { Connection } from 'typeorm';
import randomnumber from './random'


let connection: Connection;
beforeAll( async() => {
    connection = await testConnection()
})

afterAll(async() => {
    await connection.close()
})


describe('register', () => {
    it("get user by name", async() => {
        expect.assertions(4) //how many test to run
        const getUser = `
        query {
            hello(firstName: "esakki") {
                id
                firstName
                lastName
                age
                username
                email
            }
        }
        `
        //without userid
        const userAccess = await gCall({
            source: getUser,
        })
        expect(userAccess.errors?.values).toThrow()

        const result = await gCall({
            source: getUser,
            userid: "testsdsd"
        })
        expect(result).toMatchObject({
            data: {
                hello: {
                firstName: expect.any(String),
                lastName: expect.any(String),
                email: expect.any(String),
                age: expect.any(Number),
                username: expect.any(String),
              }
            }
        });
        expect(result).toBeDefined();
        expect(result.data!.hello.firstName).toBe("esakki");
    })

    it('createUser user mutation', async() => {
        expect.assertions(2) //how many test to run
        const inputUser = `
        mutation createUser($data: RegisterInput!) {
            createUser(
                data: $data
            ) {
                id
                firstName
                lastName
                email
                username
                age
            }
          }`
          const user = {
            firstName: faker.name.firstName(),
            lastName: faker.name.lastName(),
            email: faker.internet.email(),
            age: 20,
            password: faker.internet.password(),
            amount: 100
          };
          const result = await gCall({
            source: inputUser,
            variableValues: {
                data: user
            },
            userid: "testsdsd"
        })
        expect(result).toMatchObject({
            data: {
                createUser: {
                id: expect.any(String),
                firstName: expect.any(String),
                lastName: expect.any(String),
                email: expect.any(String),
                username: expect.any(String),
                age: expect.any(Number),
              }
            }
        });
        expect(result).toMatchSnapshot() //to get the result snapshot and updated --verbose in package json

    });

    it('adduser user mutation', async() => {
        expect.assertions(2) //how many test to run
        const inputUser = `
        mutation adduser(
            $firstName: String!
            $lastName: String!
            $email: String!
            $age: Float!
            $password: String!
        ) {
            adduser(
                firstName: $firstName
                lastName: $lastName
                email: $email
                age: $age
                password: $password
            ) {
                id
                firstName
                lastName
                email
                username
                age
            }
          }`
          const user = {
            firstName: faker.name.firstName(),
            lastName: faker.name.lastName(),
            email: faker.internet.email(),
            age: 20,
            password: faker.internet.password(),
            amount: 100
          };
          const result = await gCall({
            source: inputUser,
            variableValues: user,
            userid: "testsdsd"
        })
        expect(result).toMatchObject({
            data: {
                adduser: {
                id: expect.any(String),
                firstName: expect.any(String),
                lastName: expect.any(String),
                email: expect.any(String),
                username: expect.any(String),
                age: expect.any(Number),
              }
            }
        });
        expect(result).toMatchSnapshot() //to get the result snapshot and updated --verbose in package json

    });

    it("random number ", () => {
        const helpers = { randomnumber }
        helpers.randomnumber = jest.fn();
        const result = randomnumber(1, 1);
        expect(result).toBeGreaterThan(0)
        expect(result).toBeTruthy()
      });

})



