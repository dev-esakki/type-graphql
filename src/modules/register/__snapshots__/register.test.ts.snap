// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register adduser user mutation 1`] = `
Object {
  "data": Object {
    "adduser": Object {
      "age": 20,
      "email": "Gabriella_Glover@gmail.com",
      "firstName": "Timmy",
      "id": "111",
      "lastName": "Walker",
      "username": "Timmy Walker",
    },
  },
}
`;

exports[`register createUser user mutation 1`] = `
Object {
  "data": Object {
    "createUser": Object {
      "age": 20,
      "email": "Kole50@yahoo.com",
      "firstName": "Kevon",
      "id": "110",
      "lastName": "Corwin",
      "username": "Kevon Corwin",
    },
  },
}
`;
